### DISCUSSION TITLE
What is the minimum amount of RAM needed on an MCU?

### DISCUSSION THREAD

**Discussion by RaymiiOrg** [slint-ui/slint#3257] [Q&A]
The docs talk about <300 kB everywhere, but not what the minimum is. Is that known or is there a recommendation?

**Comment by tronical** [comment]
It's a little hard to say as the requirements scale with the complexity of the UI to some extent. We're running this printerdemo on the rp2040 with 260kB of RAM, and we know that if all you show is a solid filled rectangle you could go well below that. But the actual RAM requirements depend on a little on what you do.

**Comment by RaymiiOrg** [comment]
> It's a little hard to say as the requirements scale with the complexity of the UI to some extent. We're running this printerdemo on the rp2040 with 260kB of RAM, and we know that if all you show is a solid filled rectangle you could go well below that. But the actual RAM requirements depend on a little on what you do.

Thank you for the reply. As a guess, would 64 kB be too low for the solid rectangle? Or would 128 kB be better to also allow for a board runtime (network, other tasks?)

**Reply by tronical** [reply]
I think 64kb would be too low. That said, I haven't tested that.

What are your requirements?

**Reply by RaymiiOrg** [reply]
At the moment I do not have any specific requirements, just curious. The large Qt app we now have requires multiple hundreds of megabytes and were looking into alternatives which provide the same user experience and comparable developer experience. Using less resources for that same experience could mean cutting cost by using a lower specced board or even an MCU. 

But since the minimum requirements were not listed or I couldn't find them, I thought, why not just try and ask.
